components/gallery/GalleryDisplayItem.tsx

```diff
@@ -97,44 +52,52 @@ function GalleryDisplayItem(props: GalleryDisplayItemProps) {
         {spinner}
       </Col>
     );
-  } else if (pinState == PinState.PINNING) {
-    statusView = (
-      <Col className="text-white">
-        <h4>Pinning</h4>
-        {spinner}
-      </Col>
-    );
-  } else if (pinState == PinState.FAILED) {
-    statusView = (
-      <Col className="text-white mt-4">
-        <h4>Pin Failed</h4>
-      </Col>
-    );
-  } else if (pinState == PinState.NOT_FOUND) {
-    statusView = (
-      <Col className="text-white mt-4">
-        <h4>File Not Found</h4>
-      </Col>
-    );
   }
 
-  async function removeMediaGalleryItem() {
+  const removeMediaGalleryItem = React.useCallback(async () => {
+    if (!geoWebContent) {
+      return;
+    }
+
     setIsRemoving(true);
-    await mediaGalleryItemStreamManager.removeFromMediaGallery();
-    if (name) await pinningManager?.unpinCid(name);
+
+    const assetId = new AssetId({
+      chainId: `eip155:${NETWORK_ID}`,
+      assetName: {
+        namespace: "erc721",
+        reference: licenseAddress.toLowerCase(),
+      },
+      tokenId: new BN(selectedParcelId.slice(2), "hex").toString(10),
+    });
+    const ownerId = new AccountId(
+      AccountId.parse(ceramic.did?.parent.split("did:pkh:")[1] ?? "")
+    );
+    const rootCid = await geoWebContent.raw.resolveRoot({
+      ownerId,
+      parcelId: assetId,
+    });
+    const mediaGallery = await geoWebContent.raw.getPath("/mediaGallery", {
+      ownerId,
+      parcelId: assetId,
+    });
+    const newRoot = await geoWebContent.raw.putPath(
+      rootCid,
+      `/mediaGallery`,
+      mediaGallery.slice(0, index)
```

Comment:
Does this remove more than one item?

I believe you can do ```geoWebContent.raw.deletePath(rootCid, `/mediaGallery/${index}`)```

Reply:
`deletePath()` does error out with `Error: undefined is not supported by the IPLD Data Model and cannot be encoded` when passed that path.
I tried running [this test](https://github.com/Geo-Web-Project/js-geo-web/blob/main/packages/content/test/raw.test.ts#L566) with `/mediaGallery/0` (I checked that node is defined in the first place) instead of `/basicProfile/name` and it fails with the same error.

This was supposed to be a temporary workaround but `slice()` here is wrong anyhow, I must have only deleted the last inserted item while testing for this to slip through.

Reply:
`@geo-web/content@0.3.1` is published with `deletePath` fixed for arrays.

Reply:
Great! `removeMediaGalleryItem` should be fixed now that it calls `deletePath`.